packages/playground/client/src/index.ts

```diff
@@ -171,22 +170,8 @@ async function doStartPlaygroundWeb(
 	return playground;
 }
 
-const officialRemoteOrigin = 'https://playground.wordpress.net';
-function assertValidRemote(remoteHtmlUrl: string) {
-	const url = new URL(remoteHtmlUrl, officialRemoteOrigin);
-	if (
-		(url.origin === officialRemoteOrigin || url.hostname === 'localhost') &&
-		url.pathname !== '/remote.html'
-	) {
-		throw new Error(
-			`Invalid remote URL: ${url}. ` +
-				`Expected origin to be ${officialRemoteOrigin}/remote.html.`
-		);
-	}
-}
-
 function setQueryParams(url: string, params: Record<string, unknown>) {
-	const urlObject = new URL(url, officialRemoteOrigin);
+	const urlObject = new URL(url, window.location.href);
```

Comment:
That check is in place in case anyone confused `https://playground.wordpress.net/` for `https://playground.wordpress.net/remote.html`. Did it break something for you?

Reply:
Yes it broke me, in that I need to have remote.html with the same origin as my site so that I can use the origin private file system. The opfs integration is brilliant by the way. It is so much better than shared array buffer to integrate with. I would love a hint from the code on how you got that to work with the emscriptem fs by the way?????

Reply:
Oh, that check should only kick in for `playground.wordpress.net` and `localhost`, not for custom deployments. Also, this `new URL()` should work with custom domains. The second `base` argument is only used if the first argument doesn't contain a host, e.g.

```ts
> (new URL('https://mysite.com/remote.html', 'https://playground.wordpress.net')).toString()
https://mysite.com/remote.html
```

> The opfs integration is brilliant by the way. It is so much better than shared array buffer to integrate with. 

❤️ YAY, thank you!

> I would love a hint from the code on how you got that to work with the emscriptem fs by the way?????

It's a trick! :-) OPFS is asynchronous and I couldn't get it to work with Emscripten without shared array buffers, so I did this:

https://github.com/WordPress/wordpress-playground/blob/a30405b8324155ebc424eb608b4e83bb0441596b/packages/playground/sync/src/setup-playground-sync.ts

tl;dr – every MEMFS operation (create, delete, move etc) is tracked and repeated in OPFS. The "Sync changes from OPFS" button iterates through the entire OPFS handle and overwrites everything in MEMFS with OPFS data.